packages/docutils/lib/builder/nav.ts

```diff
@@ -0,0 +1,320 @@
+  const mkDocsYml = (await readYaml(mkDocsYmlPath)) as MkDocsYml;
+
+  /**
+   * Absolute path to `typedoc.json`
+   */
+  const absTypeDocJsonPath = path.isAbsolute(typeDocJsonPath)
+    ? typeDocJsonPath
+    : path.resolve(cwd, typeDocJsonPath);
+
+  /**
+   * Absolute path to TypeDoc's output directory (`out`)
+   */
+  const typeDocOutDir = path.resolve(
+    path.dirname(absTypeDocJsonPath),
+    typeDocJson.out ? typeDocJson.out : DEFAULT_REL_TYPEDOC_OUT_PATH
+  );
+
+  /**
+   * Absolute path to `mkdocs.yml`
+   */
+  const absMkdocsYmlPath = path.isAbsolute(mkDocsYmlPath)
+    ? mkDocsYmlPath
+    : path.resolve(cwd, mkDocsYmlPath);
+
+  const {docs_dir: docsDir, nav = []} = mkDocsYml;
+  /**
+   * Absolute path to the directory containing MkDocs input docs
+   */
+  const mkDocsDocsDir = path.resolve(path.dirname(absMkdocsYmlPath), docsDir ?? 'docs');
+
+  /**
+   * @todo
+   * `commands` is a dirname configurable via the `commandsDir` option added by
+   * `@appium/typedoc-plugin-appium`. this lives in `typedoc.json`, but in order for it to be parsed
+   * using TypeDoc's facilities, we have to load plugins before reading `typedoc.json`, which is
+   * slow. we will probably have to support this in the future, but for now, we can just hardcode it
+   */
+  const dirs = all ? await findDirsIn(typeDocOutDir) : [path.join(typeDocOutDir, 'commands')];
+
+  let shouldWriteMkDocsYml = false;
+
+  const navData = parseNav(nav);
+
+  // this is the thing which will be assigned to the `nav` prop
+  // and thus written to `mkdocs.yml` if there were any changes.
+  // we don't need the `name` prop, since the name is already present in the keypath.
+  const newData: Omit<ParsedNavData, 'name'>[] = [];
+
+  for (const dir of dirs) {
+    const newDataForDir: Omit<ParsedNavData, 'name'>[] = [];
+    const newRefFilepaths = await findRelativeMarkdownFiles(dir, mkDocsDocsDir);
+
+    const header = all ? _.startCase(path.basename(dir)) : DEFAULT_NAV_HEADER;
+
+    const headerItems = itemsForHeader(navData, header);
+
+    // if we found items with this header already, we are going
+    // to replace them all wholesale
+    if (headerItems.length) {
+      shouldWriteMkDocsYml = true;
+      // these are the parts of the keypath of the first item, which will contain the header string.
+      const rootHeaderKeypathParts = _.toPath(_.first(headerItems)!.keypath);
+
+      // this is the keypath up to the header string, inclusive.
+      // we append indices or names to this keypath
+      const rootHeaderKeypath = rootHeaderKeypathParts
+        .slice(0, rootHeaderKeypathParts.indexOf(header) + 1)
+        .join('.');
+
+      for (const [offset, newRefFilepath] of newRefFilepaths.entries()) {
+        const data = headerItems.find((item) => item.fileOrUrl === newRefFilepath);
+        if (data?.name) {
+          newDataForDir.push({
+            keypath: `${rootHeaderKeypath}.${offset}.${data.name}`,
```

Comment:
the condition could be simplified to `keypath: `${rootHeaderKeypath}.${offset}${data.name ? ('.' + data.name) : ''}`,`

Reply:
I didn't want to embed a ternary in a template, I guess.